src/authn/index.js

```diff
@@ -210,36 +276,26 @@ function setup(app) {
      *   -trs, 18 March 2022
      */
 
-    return done(null, user);
-  });
+    /* Check the final user object is the right shape.  This would be better
+     * with types and TypeScript, or even a validation library, but asserts are
+     * ok for now in the absence of either of those.
```

Comment:
"Low on the priority list" is the primary reason I haven't done/pushed for it yet, though I've gotten close to JFDI a few times. One tempering force has been that while it might be a 1-2 day project to get the _ability_ to use TypeScript for the server, there will be a long tail of untyped JS that will require using the Any type in many places for a while, and that erodes a lot of the typing benefits (but certainly not all).

@jameshadfield may have thoughts here, as he's preferred (at least historically) not having a build/compilation step for the server here.